### PR TITLE
Replace termion with crossterm

### PR DESCRIPTION
**Pull request by jtschuster**
This isolates the changes from https://github.com/cantino/mcfly/pull/280 that are related to crossterm.

@cantino Hopefully this makes it easier to review and test yourself, and PowerShell/Windows support can be added separately.

### CODE REVIEW

**Comment by cantino** [comment]
Is this ready for testing @jtschuster?

**Comment by jtschuster** [comment]
@cantino Sorry for the delay, it should be ready to look at now.

**Review by cantino — CHANGES_REQUESTED** [review]
I get 

```
 $ thread 'main' panicked at 'McFly error: failed to read input IoError(Custom { kind: Other, error: "Failed to initialize input reader" })', src/interface.rs:413:43
 git st                                             note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                           2h 11m
```

On my Mac when I run this.

File: src/interface.rs [hunk 1]
```diff
@@ -412,123 +397,229 @@ impl<'a> Interface<'a> {
...
+        .unwrap();
     }
 
-    fn select_with_emacs_key_scheme(&mut self, k: Key) -> bool {
-        match k {
-            Key::Char('\n') | Key::Char('\r') | Key::Ctrl('j') => {
+    fn select_with_emacs_key_scheme(&mut self, event: KeyEvent) -> bool {
+        match event {
+            KeyEvent {
+                code: KeyCode::Enter,
```
[the remark this answers is not in the document]
Comment (jtschuster) [reply src/interface.rs:474]: I thought Enter would cover both, but I'll add \r and \n to be safe.

**Comment by jtschuster** [comment]
It looks like there is a known issue with crossterm on mac when /dev/tty is used instead of stdin. Since it looks like zsh's zle uses the tty, that seems to be causing issues. It's working for me on linux with bash and zsh, and mac with bash, but not zsh.
There's a couple PR's looking to address the issue, one stale and one new:
https://github.com/crossterm-rs/crossterm/pull/407
https://github.com/crossterm-rs/crossterm/pull/711

**Comment by jtschuster** [comment]
@cantino Sorry for letting this get stale, but it should be ready to take another look at now. The crashing on zsh issue was fixed in crossterm v0.26, and it seems to be working for me. Please take a look when you can.

**Comment by cantino** [comment]
Thank you for your hard work on this @jtschuster. Have you been using this personally without issue?

**Comment by jtschuster** [comment]
> Thank you for your hard work on this @jtschuster. Have you been using this personally without issue?

I use windows primarily, but I've been using https://github.com/cantino/mcfly/pull/280 without issues so far, and haven't hit issues testing this branch yet.

> Minor: The highlight color is a bit hard to read on my Mac on iTerm2:

I just pushed a commit that changes the highlight color to DarkGreen and it's much more readable.

> Is now a good time to allow color customization?

I wouldn't want to add too much to this PR, but it should be easy to rebase https://github.com/cantino/mcfly/pull/156 once this is in.

**Comment by cantino** [comment]
In light mode, the text you've typed isn't visible when selected:

image

While it is visible in dark mode:

image

And it is on master:
image

**Comment by jtschuster** [comment]
Good catch! I've pushed a commit that makes it match the behavior in master.

**Comment by praveenperera** [comment]
Thanks for the or @jtschuster, i’m going to test this out locally tomorrow. For now would you find fixing the last clippy warnings?

**Comment by praveenperera** [comment]
@jtschuster @cantino I've been using this branch, its been good

**Comment by cantino** [comment]
One more merge and I think we're good to go on this.

**Review by cantino — APPROVED** [review]

**Comment by cantino** [comment]
Thanks @jtschuster! Awesome work.

**Comment by cantino** [comment]
Released in v0.8.0.

**Pull request merged**